content/posts/irods-development-update-july-2024.md

```diff
@@ -0,0 +1,273 @@
+
+For the iRODS server, a regression in the replication implementation has been fixed. This change restores the users' ability to replicate data when they only have permissions via a group. `trimPrefix` and `trimStrings` _(C functions provided by the iRODS library)_ are now deprecated. Here's a big one, the next release of iRODS will officially support Ubuntu 24.04! We're getting better and better at adding support for new operating systems. Aside from that, we've added documentation for new versions of PostgreSQL and MySQL. Lastly, a new GenQuery1 parser has been implemented. The new parser preserves all of the capabilities and features of GenQuery1, but adds the ability to escape characters and bytes, just like in GenQuery2. Given that GenQuery2 is very different from GenQuery1, we wanted to see if we could offer a parser which focused on the immediate issues, embedded single quotes and keywords in string literals. We're unsure whether this work will ever be merged into iRODS, but if you're interested in following along, take a look at [https://github.com/irods/irods/pull/7819](https://github.com/irods/irods/pull/7819).
+
+Users of the iRODS Testing Environment will be happy to know that projects for PostgreSQL 16, MySQL 8.4, and MariaDB 10.6, 10.11, and 11.4 are now available.
+
+The GenQuery python module _(i.e. genquery.py)_ which ships with the Python rule engine plugin is gaining support for GenQuery2. Once complete, the `Query` class will give users the ability to choose the GenQuery parser they'd like to use. With these changes, administrators will have an easy migration path from GenQuery1 to GenQuery2. This is especially helpful in situations where the policy makes heavy use of the `Query` class.
+
+The S3 resource plugin received a couple fixes. The first fix resolves an issue where the server was not honoring the number of threads specified by `iput -N`. The second fix resolves an issue with `irsync` and checksum operations failing.
+
+For the Python iRODS client (PRC), session objects now propagate connection timeout information to the socket layer as originally intended. Support for progress bars has been improved. Users will soon be able to use their favorite progress bar library with the PRC. And last but not least, a question about the recently published ISO standard, ISCC, was presented to the community. If you find that interesting and/or have suggestions, please see [https://github.com/irods/python-irodsclient/issues/573](https://github.com/irods/python-irodsclient/issues/573). The issue also provides a link to an implementation of ISCC for the PRC!
+
+For the Automated Ingest tool, we've adjusted the event handler interface to account for changes introduced in an unreleased update. This is a **breaking** change. Users of the the Automated Ingest tool will need to update their event handlers. We've also added a Docker Compose project which makes it possible to easily test the S3 scanner. 
```

Comment:
we actually released 0.5.0 with this set of changes - in case you want to mention that.

Reply:
Reworded that line. Let me know if it needs to be tweaked.